pygitea_mod/gitea/gitea.py

```diff
@@ -157,6 +159,14 @@ def requests_patch(self, endpoint: str, data: dict):
             self.logger.error(error_message)
             raise Exception(error_message)
         return self.parse_result(request)
+    
+    def requests_patch_mod(self, endpoint: str, data: dict):
+        request = self.requests.patch(self.__get_url(endpoint), headers=self.headers, data=json.dumps(data))
+        if request.status_code not in [200, 201]:
+            error_message = f"Received status code: {request.status_code} ({request.url}) {data}"
+            self.logger.error(error_message)
+            raise Exception(error_message)
```

Comment:
Log-and-raise is an anti-pattern.  Do one or the other.  (I would probably raise.)  If you do both, presumably something else at a higher level is going to catch the exception and do something with it, probably print or log it.  So you end up with duplicates.

Reply:
Thank you. I have all sorts of questions on error checking, try/except blocks.